.travis.yml

```
@@ -1,86 +1,71 @@
-    android:
-      components:
-      - tools
-      - build-tools-26.0.2
-  - env: PLATFORM=android-6.0
-    os: linux
-    language: android
-    jdk: oraclejdk8
-    android:
-      components:
-      - tools
-      - build-tools-26.0.2
-  - env: PLATFORM=android-7.0
-    os: linux
-    language: android
-    jdk: oraclejdk8
-    android:
-      components:
-      - tools
-      - build-tools-26.0.2
+    - env: PLATFORM=browser-chrome
+    - env: PLATFORM=browser-firefox
+    - env: PLATFORM=browser-safari
+    - env: PLATFORM=browser-edge
+    - env: PLATFORM=ios-9.3
+      os: osx
+      osx_image: xcode7.3
+    - env: PLATFORM=ios-10.0
+      os: osx
+      osx_image: xcode7.3
+    - env: PLATFORM=android-4.4
+      language: android
+      jdk: oraclejdk8
+      android:
+        components:
+          - tools
+          - build-tools-26.0.2
+    - env: PLATFORM=android-5.1
+      language: android
+      jdk: oraclejdk8
+      android:
+        components:
+          - tools
+          - build-tools-26.0.2
+    - env: PLATFORM=android-6.0
+      language: android
+      jdk: oraclejdk8
+      android:
+        components:
+          - tools
+          - build-tools-26.0.2
+    - env: PLATFORM=android-7.0
+      language: android
+      jdk: oraclejdk8
+      android:
+        components:
+          - tools
+          - build-tools-26.0.2
+
 before_install:
-- rm -rf ~/.nvm && git clone https://github.com/creationix/nvm.git ~/.nvm && (cd ~/.nvm
-  && git checkout `git describe --abbrev=0 --tags`) && source ~/.nvm/nvm.sh && nvm
-  install $TRAVIS_NODE_VERSION
-- node --version
-- if [[ "$PLATFORM" =~ android ]]; then gradle --version; fi
-- if [[ "$PLATFORM" =~ ios ]]; then npm install -g ios-deploy; fi
-- if [[ "$PLATFORM" =~ android ]]; then echo y | android update sdk -u --filter android-22,android-23,android-24,android-25,android-26,android-27;
-  fi
-- git clone https://github.com/apache/cordova-paramedic /tmp/paramedic && pushd /tmp/paramedic
-  && npm install && popd
-- npm install -g cordova
+  # `language: android` has no Node.js installed, therefore we need to install it manually
+  - if [[ "$PLATFORM" =~ android ]]; then nvm install $TRAVIS_NODE_VERSION; fi
+  - node --version
+  - if [[ "$PLATFORM" =~ android ]]; then gradle --version; fi
+  - if [[ "$PLATFORM" =~ ios ]]; then npm install -g ios-deploy; fi
+  - if [[ "$PLATFORM" =~ android ]]; then echo y | android update sdk -u --filter android-22,android-23,android-24,android-25,android-26,android-27; fi
+  - npm install -g cordova-paramedic@https://github.com/apache/cordova-paramedic.git
+  - npm install -g cordova
+
 install:
-- npm install
+  - npm install
+
 script:
-- npm test
-- node /tmp/paramedic/main.js --config pr/$PLATFORM --plugin $(pwd) --shouldUseSauce
-  --buildName travis-plugin-statusbar-$TRAVIS_JOB_NUMBER
+  - npm test
+  - cordova-paramedic --config pr/$PLATFORM --plugin $(pwd) --shouldUseSauce --buildName travis-plugin-statusbar-$TRAVIS_JOB_NUMBER
```

```suggestion
  - cordova-paramedic --config pr/$PLATFORM --plugin $(pwd) --shouldUseSauce --buildName travis-$(pwd)-$TRAVIS_JOB_NUMBER
```

Would this work? Then this line would be generalized as well and could directly be copy pasted to other plugins.

---

I believe not. pwd is equivalent `$TRAVIS_BUILD_DIR´ (e.g.
`/home/travis/build/timbru31/cordova-plugin-statusbarto`)


We could use $TRAVIS_REPO_SLUG (e.g. `apache/cordova-plugin-statusbar`)

---

Ok, let's see if Saucelabs likes the `/`.

---

Change is pushed, let's wait for Travis: https://travis-ci.org/apache/cordova-plugin-statusbar/builds/486645937

---

Seems to work: https://app.saucelabs.com/tests/2797961f26884526aa8bae754c13fd8d/commands#3 🎉

---

Can you see if the test/session was created at SauceLabs? Travis terminated the job because of:
```
cordova-paramedic: Starting Sauce Connect...
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

---

No, it was not. Only build numbers 261.2 and up :/

---

Another thing I just noticed on Travis: ` Node.js: $TRAVIS_NODE_VERSION`. Not very insightful. Not a big problem now with just 1 version, but may be later if we decide to also test multiple :(

---

Restarted

---

Definitely not in this PR - this would best be a new discussion issue in /cordova first.

My quick thoughts anyway:
In general: Sure, why not. The limiting factor here is test run time. The Apache CI environments on Travis and AppVeyor are very crowded, so it might happen that one has to wait pretty long. Longer test run time will increase that problem.

Also: Does it make sense to run all tests on multiple node versions? Does this affect what paramedic does (and especially on SauceLabs) at all?

---

Alright, I guess https://github.com/apache/cordova/issues/37 is already a good candidate to discuss the CI with more Node.js versions.  


> Also: Does it make sense to run all tests on multiple node versions? Does this affect what paramedic does (and especially on SauceLabs) at all?

SauceLabs tests the real browsers, so using more Node.js versions would (hopefully) not change the behavior or output. The tests would more or less test paramedic itself which is run by Node.js. Maybe we could use always the latest LTS release of Node.js in the plugins (reduces maintenance) CI config and test paramedic against multiple Node.js versions.